pkg/controllers/override/imagepatch.go

```diff
@@ -0,0 +1,253 @@
+			}
+		} else {
+			if err := parsePatchesFromWorkload(fedObject, imageOverrider, patchMap); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// convert patchMap to patchList and return
+	patches := make(fedcorev1a1.OverridePatches, len(patchMap))
+	index := 0
+	for path, value := range patchMap {
+		patches[index] = fedcorev1a1.OverridePatch{
+			Op:   OperatorReplace,
+			Path: path,
+			// OverridePatch's value is apiextensionsv1.JSON type, which raw should be wrapped with "".
+			Value: apiextensionsv1.JSON{Raw: []byte(fmt.Sprintf("\"%s\"", value))},
+		}
+		index++
+	}
+
+	// sort by path to avoid unnecessary updates due to disorder
+	sort.Slice(patches, func(i, j int) bool {
+		return patches[i].Path < patches[j].Path
+	})
+
+	return patches, nil
+}
+
+// parsePatchesFromImagePath applies overrider to image value on the specified path,
+// and generates a patch which is stored in patchMap
+func parsePatchesFromImagePath(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	imageOverrider *fedcorev1a1.ImageOverrider,
+	patchMap map[string]string,
+) error {
+	imagePath := imageOverrider.ImagePath
+	if !strings.HasPrefix(imagePath, pathSeparator) {
+		return fmt.Errorf("image path should be start with %s", pathSeparator)
+	}
+	imageValue := patchMap[imagePath]
+
+	// get image value
+	if imageValue == "" {
+		// get source obj
+		sourceObj, err := fedObject.GetSpec().GetTemplateAsUnstructured()
+		if err != nil {
+			return fmt.Errorf("failed to get sourceObj from fedObj: %w", err)
+		}
+
+		imageValue, err = GetStringFromUnstructuredObj(sourceObj, imageOverrider.ImagePath)
+		if err != nil {
+			return fmt.Errorf("failed to parse image value from unstructured obj: %w", err)
+		}
+	}
+
+	// apply image override
+	newImageValue, err := applyImageOverride(imageValue, imageOverrider)
+	if err != nil {
+		return fmt.Errorf("failed to apply image override: %w", err)
+	}
+	patchMap[imagePath] = newImageValue
+
+	return nil
+}
+
+// parsePatchesFromWorkload applies overrider to image value on the default path of workload,
+// and generates a patch which is stored in patchMap
+func parsePatchesFromWorkload(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	imageOverrider *fedcorev1a1.ImageOverrider,
+	patchMap map[string]string,
+) error {
+	// get pod spec from fedObj
+	sourceObj, err := fedObject.GetSpec().GetTemplateAsUnstructured()
+	if err != nil {
+		return fmt.Errorf("failed to get sourceObj from fedObj: %w", err)
+	}
+	gvk := sourceObj.GetObjectKind().GroupVersionKind()
+	podSpec, err := podutil.GetResourcePodSpec(fedObject, gvk)
+	if err != nil {
+		return fmt.Errorf("failed to get podSpec from sourceObj: %w", err)
+	}
+
+	// convert user-specified containerNames into set
+	specifiedContainers := sets.New[string]()
+	for _, name := range imageOverrider.ContainerNames {
+		if name != "" {
+			specifiedContainers.Insert(name)
+		}
+	}
+
+	// process the init containers
+	for index, container := range podSpec.InitContainers {
```

Comment:
```suggestion
    for _, containers := range [][]corev1.Container{podSpec.InitContainers, podSpec.Containers} {
		for index, container := range containers {
```

Duplication is unhealthy, and can be avoided like this.

Reply:
The code has been updated.